app/configuration/api.go

```diff
@@ -58,7 +59,15 @@ func (srv *Service) getFileMetadataFromAPI(ctx context.Context, src string) (*Fi
 	fileMetadataResp := new(fileMetadataResponse)
 
 	if err := srv.api.Get(ctx, path, fileMetadataResp); err != nil {
-		return nil, fmt.Errorf("error getting file metadata: %w", err)
+
+		asErr, ok := err.(*api.Error)
+
+		if ok && asErr.ResponseCode >= http.StatusBadRequest {
+			return nil, fmt.Errorf("error getting file metadata: %w", err)
+		}
+
+		// Do not return if there's no http error (e.g. during network outage)
+		return nil, nil
```

Comment:
I don't think we should return `nil` error in an error case. If the goal is to not log connectivity errors, we should handle that inside report recording function (`ReportError(ctx context.Context, extraLog any, msgFmt string, args ...any)` <- we can check the `extraLog`) and check whether `errors.Is(err, api.ConnectionError)` or even the old-school way `if _, skipLogging = extraLog.(api.ConnectionError); skipLogging { return }`

Reply:
That seems like a way better approach. I'll close this PR and make a new one.